File: app/src/main/java/co/ab180/awesomeapp/ui/main/MainActivity.java
package co.ab180.awesomeapp.ui.main;

import android.app.Activity;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import co.ab180.airbridge.Airbridge;
import co.ab180.airbridge.event.ecommerce.HomeViewEvent;
import co.ab180.awesomeapp.R;
import co.ab180.awesomeapp.domain.model.Product;
import co.ab180.awesomeapp.network.ApiService;

public class MainActivity extends Activity {

    private ProductsAdapter adapter;
    private RecyclerView recyclerView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);
        adapter = new ProductsAdapter();

        initContents();
    }

    @Override
    protected void onResume() {
        super.onResume();
        sendEvent();
    }

    private void initContents() {
        List<Product> products = ApiService.INSTANCE.getProducts(this);
        adapter.submitList(products);
        recyclerView.setAdapter(adapter);
    }

    private void sendEvent() {
        HomeViewEvent event = new HomeViewEvent();
        Airbridge.trackEvent(event);
    }
}

File: app/src/main/java/co/ab180/awesomeapp/ui/main/MainActivity.kt
package co.ab180.awesomeapp.ui.main

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import co.ab180.awesomeapp.R
import co.ab180.awesomeapp.domain.model.Product
import co.ab180.awesomeapp.network.ApiService
import com.google.gson.Gson
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private val adapter = ProductsAdapter()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initContents()
    }

    private fun initContents() {
        val products = ApiService.getProducts(this)
        adapter.submitList(products)
        recyclerView.adapter = adapter
    }
}
